src/main/java/backend/resource/TurboIssue.java

```diff
@@ -334,17 +329,35 @@ public IssueMetadata getMetadata() {
     public void setMetadata(IssueMetadata metadata) {
         this.metadata = metadata;
     }
+
     public Optional<LocalDateTime> getMarkedReadAt() {
         return markedReadAt;
     }
+
     public void setMarkedReadAt(Optional<LocalDateTime> markedReadAt) {
         this.markedReadAt = markedReadAt;
     }
+
     public boolean isCurrentlyRead() {
-        return isCurrentlyRead;
+        if (!getMarkedReadAt().isPresent()) {
+            return false;
+        }
+        if (getUpdatedAt() == null) {
```

Comment:
Is the update time ever null? If this is to account for GitHub Issues that are just created, maybe we could set it to the creation time in the TurboIssue constructor.